pkg/controller/metrics/provision_underway_collector.go

```diff
@@ -96,14 +96,124 @@ func (cc provisioningUnderwayCollector) Describe(ch chan<- *prometheus.Desc) {
 	prometheus.DescribeByCollect(cc, ch)
 }
 
-func newProvisioningUnderwayCollector(client client.Client) prometheus.Collector {
+var (
+	metricClusterDeploymentProvisionUnderwaySecondsDesc = prometheus.NewDesc(
+		"hive_cluster_deployment_provision_underway_seconds",
+		"Length of time a cluster has been provisioning.",
+		[]string{"cluster_deployment", "namespace", "cluster_type", "condition", "reason", "platform", "image_set"},
+		nil,
+	)
+)
+
+func newProvisioningUnderwayCollector(client client.Client, minimum time.Duration) prometheus.Collector {
 	return provisioningUnderwayCollector{
 		client: client,
-		metricClusterDeploymentProvisionUnderwaySeconds: prometheus.NewDesc(
-			"hive_cluster_deployment_provision_underway_seconds",
-			"Length of time a cluster has been provisioning.",
-			[]string{"cluster_deployment", "namespace", "cluster_type", "condition", "reason", "platform", "image_set"},
-			nil,
-		),
+		metricClusterDeploymentProvisionUnderwaySeconds: metricClusterDeploymentProvisionUnderwaySecondsDesc,
+		minDuration: minimum,
+	}
+}
+
+// provisioning underway install restarts metrics collected through a custom prometheus collector
+type provisioningUnderwayInstallRestartsCollector struct {
+	client client.Client
+
+	// minRestarts, when non-zero, is the minimum restarts after which clusters provisioning
+	// will start becoming part of the metric. When set to zero, all clusters provisioning
+	// will be included in the metric.
+	minRestarts int
+
+	// metricClusterDeploymentProvisionUnderwayInstallRestarts is a prometheus metric for the number of install
+	// restarts for a still provisioning cluster.
+	metricClusterDeploymentProvisionUnderwayInstallRestarts *prometheus.Desc
+}
+
+// collects the metrics for provisioningUnderwayCollector
```

Comment:
This doc is now confusing. We can even scrape it since the func definition is enough to understand